x-pack/plugins/alerting/server/authorization/alerting_authorization_kuery.ts

```diff
@@ -35,22 +37,40 @@ const esQueryConfig: EsQueryConfig = {
 
 export function asFiltersByRuleTypeAndConsumer(
   ruleTypes: Set<RegistryAlertTypeWithAuth>,
-  opts: AlertingAuthorizationFilterOpts
+  opts: AlertingAuthorizationFilterOpts,
+  includeSpaceId = false,
+  spaceId?: string | undefined
 ): KueryNode | JsonObject {
   const kueryNode = nodeBuilder.or(
     Array.from(ruleTypes).reduce<KueryNode[]>((filters, { id, authorizedConsumers }) => {
       ensureFieldIsSafeForQuery('ruleTypeId', id);
-      filters.push(
-        nodeBuilder.and([
-          nodeBuilder.is(opts.fieldNames.ruleTypeId, id),
-          nodeBuilder.or(
-            Object.keys(authorizedConsumers).map((consumer) => {
-              ensureFieldIsSafeForQuery('consumer', consumer);
-              return nodeBuilder.is(opts.fieldNames.consumer, consumer);
-            })
-          ),
-        ])
-      );
+      if (includeSpaceId && opts.fieldNames.spaceIds != null && spaceId != null) {
```

Comment:
What about something like this to reduce code duplication:

```
let andNodes = [
  nodeBuilder.is(opts.fieldNames.ruleTypeId, id),
  nodeBuilder.or(
    Object.keys(authorizedConsumers).map((consumer) => {
      ensureFieldIsSafeForQuery('consumer', consumer);
      return nodeBuilder.is(opts.fieldNames.consumer, consumer);
    })
  ),
];

if (includeSpaceId && opts.fieldNames.spaceIds != null && spaceId != null) {
  andNodes.push(nodeBuilder.is(opts.fieldNames.spaceIds, spaceId));
}

filters.push(
  nodeBuilder.and(andNodes)
);
```

Reply:
Yea, def. I updated the code.